code/datums/weather/weather.dm

```diff
@@ -105,14 +105,16 @@
 	weather_duration = rand(weather_duration_lower, weather_duration_upper)
 	START_PROCESSING(SSweather, src)
 	update_areas()
-	for(var/M in GLOB.player_list)
+	addtimer(CALLBACK(src, .proc/start), telegraph_duration)
+	for(var/mob/M in GLOB.player_list)
 		var/turf/mob_turf = get_turf(M)
 		if(mob_turf && (mob_turf.z in impacted_z_levels))
 			if(telegraph_message)
 				to_chat(M, telegraph_message)
+			if(M?.client?.prefs?.toggles_sound & SOUND_WEATHER)
+				return
```

Comment:
because this should be a continue not a return

Reply:
Done